### PR TITLE
Revamp symbols page

### PR DESCRIPTION
**Pull request by janfaracik**
symbols

Felt that this page could do with a bit of love, also enabled some thinking/building of components to use across the library in the future.

**TODO**
- [x] Classes aren't working on Symbols, this'll require a Jenkins fix I imagine
- [x] I18N
- [x] Add more symbols to the Java method so that the masthead doesn't repeat symbols

Theme aware code blocks would be nice (https://github.com/jenkinsci/design-library-plugin/issues/68)

<a href="https://gitpod.io/#https://github.com/jenkinsci/design-library-plugin/pull/67"></a>

### CODE REVIEW

**Comment by timja** [comment]
> * Theme aware code blocks would be nice

would be nice but not related to this PR :)

created https://github.com/jenkinsci/design-library-plugin/issues/68

**Comment by timja** [comment]
I tried it out it in gitpod and it looks really nice :)

**Comment by timja** [comment]
It would be good to bump the minimum Jenkins version too I think

**Review by timja — COMMENTED** [review]
I tested the last commits in gitpod.

2 comments both non blocking

1. contrast is not great with the dark theme changes, e.g. on l:icon
2. do the symbols have to be sooo large in the examples? Seems over the top to me and makes lots of scrolling


image

**Comment by lemeurherve** [comment]
The size of the symbols are wrong on preview https://8080-janfaracik-designlibrar-yznb1gxl2iz.ws-eu45.gitpod.io/jenkins/design-library/Symbols/ :

image

**Comment by timja** [comment]
In safari some don't seem to render:

image

**Comment by janfaracik** [comment]
Those crazy icon sizes are definitely not intentional :P Jenkins doesn't seem to apply classes correctly hence why they're ballooning in size. Changing `classSpec` to `attrs.class` in `jenkins/icon.jelly` fixes this, but I haven't tested if that causes issues elsewhere in Jenkins.

image

**Comment by janfaracik** [comment]
Should be good for reviews once https://github.com/jenkinsci/jenkins/pull/6609 is merged in :)

File: pom.xml [hunk 1]
```diff
@@ -44,7 +44,7 @@ THE SOFTWARE.
     <properties>
         <changelist>999999-SNAPSHOT</changelist>
         <gitHubRepo>jenkinsci/design-library-plugin</gitHubRepo>
-        <jenkins.version>2.341</jenkins.version>
+        <jenkins.version>2.348</jenkins.version>
```
Comment (timja) [inline pom.xml:47]:
```suggestion
        <jenkins.version>2.350-rc32407.d15ca_b_52160b_</jenkins.version>
```

File: pom.xml [hunk 2]
```diff
@@ -44,7 +44,7 @@ THE SOFTWARE.
     <properties>
         <changelist>999999-SNAPSHOT</changelist>
         <gitHubRepo>jenkinsci/design-library-plugin</gitHubRepo>
-        <jenkins.version>2.341</jenkins.version>
+        <jenkins.version>2.350-rc32407.d15ca_b_52160b_</jenkins.version>
```
Comment (timja) [inline pom.xml:47]: TODO switch to new version once released

**Review by NotMyFault — APPROVED** [review]
Great work, I like the symbol pop a lot :P

File: src/main/resources/scss/abstracts/_typography.scss [hunk 3]
```diff
@@ -1,21 +1,41 @@
...
 
   & + & {
-    margin-top: calc((var(--section-padding) / 2) * -1);
+    margin-top: calc((var(--jdl-spacing) / 2) * -1);
   }
 }
 
+.jdl-list {
+  font-size: 1.1rem;
+  font-weight: 500;
```
Comment (timja) [inline src/main/resources/scss/abstracts/_typography.scss:31]: is this font-weight needed?

it seems a bit much in:

image

**Review by timja — APPROVED** [review]
Looks amazing

File: src/main/resources/scss/abstracts/_typography.scss (hunk 3 above)
Comment (janfaracik) [reply src/main/resources/scss/abstracts/_typography.scss:31]: Are you on Chrome? I've just opened a PR (that you've just approved :P) https://github.com/jenkinsci/jenkins/pull/6614#pullrequestreview-989694771 to address the weird font issues on there. Fonts should have better weighting now on Chrome, be that if they're light or bold.

https://user-images.githubusercontent.com/43062514/171062827-c318c45c-abf3-4159-a04d-7fb73a20654c.mov

Comment (timja) [reply src/main/resources/scss/abstracts/_typography.scss:31]: Yes on chrome

Comment (janfaracik) [reply src/main/resources/scss/abstracts/_typography.scss:31]: Is the new font rendering better for you now?

File: pom.xml (hunk 2 above)
Comment (timja) [inline pom.xml:47]:
```suggestion
        <jenkins.version>2.350-rc32411.186041cf1c8b_</jenkins.version> <!-- TODO https://github.com/jenkinsci/jenkins/pull/6614 -->
```

File: src/main/resources/scss/abstracts/_typography.scss (hunk 3 above)
Comment (timja) [reply src/main/resources/scss/abstracts/_typography.scss:31]: Still a bit bold for regular text I think,

image

**Pull request merged**